File: ios/BearoundReactSdkEventEmitter.m
#import "BearoundReactSdkEventEmitter.h"

static __weak BearoundReactSdkEventEmitter *sharedEmitter = nil;

@implementation BearoundReactSdkEventEmitter

RCT_EXPORT_MODULE(BearoundReactSdkEventEmitter)

+ (void)emit:(NSString *)name body:(id)body
{
  if (sharedEmitter == nil) {
    return;
  }
  [sharedEmitter sendEventWithName:name body:body];
}

- (instancetype)init
{
  self = [super init];
  if (self) {
    sharedEmitter = self;
  }
  return self;
}

- (NSArray<NSString *> *)supportedEvents
{
  return @[
    @"bearound:beacons",
    @"bearound:sync",
    @"bearound:scanning",
    @"bearound:error"
  ];
}

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/BearoundReactSdkEventEmitter.swift
import Foundation
import React

@objc(BearoundReactSdkEventEmitter)
public class BearoundReactSdkEventEmitter: RCTEventEmitter {
  private static var shared: BearoundReactSdkEventEmitter?

  public override init() {
    super.init()
    BearoundReactSdkEventEmitter.shared = self
  }

  @objc public static func emit(_ name: String, body: Any) {
    shared?.sendEvent(withName: name, body: body)
  }

  public override func supportedEvents() -> [String]! {
    return [
      "bearound:beacons",
      "bearound:syncLifecycle",
      "bearound:backgroundDetection",
      "bearound:scanning",
      "bearound:error",
      // v2.4 — beacon region lifecycle
      "bearound:beaconRegion",
      "bearound:activeScan",
      // v2.5 — Bluetooth "two eyes" zone (iOS-only)
      "bearound:bluetoothZone",
      "bearound:bluetoothScanMode",
      // Bluetooth adapter state (poweredOn/off/unauthorized/...)
      "bearound:bluetoothState",
    ]
  }

  @objc public override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
